routing/router.go

```diff
@@ -904,7 +908,16 @@ func generateSphinxPacket(route *Route, paymentHash []byte) ([]byte, error) {
 		}),
 	)
 
-	return onionBlob.Bytes(), nil
+	// Initialize the onion error deobfuscator in order to be able to decode
```

Comment:
In line with the comment above, a `sphinx.Circuit` should be created and returned here instead.